### PR TITLE
[Rule Request]:  Prohibiting use of aria-hidden and role='presentation' on focusable elements.

### LINKED ISSUE #85: [Rule Request] Rule for prohibiting use of `aria-hidden` or `role=presentation` on focusable elements
As this breaks the forth rule of aria :D

Reference:
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute
- https://www.w3.org/TR/using-aria/#fourth

### PR DESCRIPTION
**Pull request by mauryapari**
- Closes #85 
- Added rule for prohibiting use of aria-hidden and role='presentation' on focusable elements.
- Add checks to prohibit use of aria-hidden and role='presentation' on parents of focusable elements.

### CODE REVIEW

**Comment by mauryapari** [comment]
@kddnewton ,  @vhoyer. Could you please reiview this PR? Any feedback you provide would be greatly appreciated.

**Comment by vhoyer** [comment]
Hey, could you add a new line at the end of the files that do not have them? github adds a little :no_entry_sign: at the end of the file when that's the case

**Review by vhoyer — APPROVED** [review]
code wise it seems fine, I just don't like nested ifs if I can avoid it, but not a real problem

**Comment by vhoyer** [comment]
but prettier is complaining about your formatting, can you pls fix the problems before we merge it?

**Comment by mauryapari** [comment]
Hey @vhoyer. fixed both of them. do check it.

**Comment by vhoyer** [comment]
the docs both seem to still have the final EOL problem

**Comment by mauryapari** [comment]
Hey. not sure what happened earlier. i have added it again for those.

**Comment by vhoyer** [comment]
```
 $ /home/runner/work/eslint-plugin-vuejs-accessibility/eslint-plugin-vuejs-accessibility/node_modules/.bin/prettier --check '**/*'
Checking formatting...
[warn] docs/rules/no-aria-hidden-on-focusable.md
[warn] docs/rules/no-role-presentation-on-focusable.md
[warn] src/rules/__tests__/no-aria-hidden-on-focusable.test.ts
[warn] src/rules/__tests__/no-role-presentation-on-focusable.test.ts
[warn] src/rules/no-aria-hidden-on-focusable.ts
[warn] src/rules/no-role-presentation-on-focusable.ts
[warn] src/utils/hasFocusableElement.ts
[warn] Code style issues found in 7 files. Run Prettier with --write to fix.```

**Comment by mauryapari** [comment]
okay i believe it should be resolved now. have a look.

File: src/rules/no-role-presentation-on-focusable.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import type { Rule } from "eslint";
+// import type { AST } from "vue-eslint-parser";
```
Comment (vhoyer) [inline src/rules/no-role-presentation-on-focusable.ts:2]: it would be better if we removed the comments, no?

**Review by vhoyer — APPROVED** [review]

File: src/rules/no-role-presentation-on-focusable.ts (hunk 1 above)
Comment (mauryapari) [reply src/rules/no-role-presentation-on-focusable.ts]: done

**Review by vhoyer — APPROVED** [review]

**Pull request merged**